src/Drivers/CMakeLists.txt

```
@@ -39,5 +45,5 @@ endif()
 
 if(HIOP_BUILD_SHARED AND NOT HIOP_USE_GPU)
```

```suggestion
if(HIOP_BUILD_SHARED)
```
Do we need to disable shared library build when enabling GPU support? I believe this is remnant of an older version of build system.

@cnpetra, this may cause the test failure you observed.

---

I think you're entirely correct - @cnpetra can you confirm?

---

In short, I am not sure... I do not recall whether the shared library does not work on GPUs or there are other limitation(s) preventing enabling shared library for GPUs. @michel2323 maybe you can shed some light? 

In any case, it looks like we need a C RAJA using shared library example in the near future to ensure that the latest bells and whistles in HiOp work with C interface and shared library (which are the preamble for interfacing with Julia)

---

Nothing that I remember. I only remember the CMake issues. A while ago I tried HiOp using CUDA internally and it's all good so far, even through Julia.